src/GraphQL/Types/Composite/ComplexGraphType.cs

```diff
@@ -97,13 +97,13 @@ public virtual FieldType AddField(FieldType fieldType)
             {
                 if (this is IInputObjectGraphType)
                 {
-                    if (fieldType.ResolvedType?.IsInputType() == false || fieldType.Type?.IsInputType() == false)
+                    if (fieldType.ResolvedType != null ? fieldType.ResolvedType.IsInputType() == false : fieldType.Type?.IsInputType() == false)
```

Comment:
I don't undestand the difference.

Reply:
If `ResolvedType` contains an object, `Type` is ignored

Reply:
If `fieldType.ResolvedType` is `null` then `fieldType.ResolvedType?.IsInputType()` is `null` too and `null` == `false` yields `false`.

Reply:
The same below.

Reply:
That's not the part of the equation that has changed.  The point is to skip checking `Type` if `ResolvedType` contains an object, whether it is compatible or not, since `Type` is not used in that circumstance.

Reply:
> If ResolvedType contains an object, Type is ignored

So it happened. This is bad? After changes I see the same behavior.

Reply:
Yes, that's what fixes the issue posted by OpenSpacesAndPlaces in #1927 .  He had set both `ResolvedType` and `Type`.

Reply:
See https://github.com/graphql-dotnet/graphql-dotnet/issues/1927#issuecomment-774198841

Reply:
I added tests to verify that code like his would work correctly now.  See the test I wrote, which would fail without this change.  Notice this line of code:

```cs
obj.AddField(new FieldType { Name = "list", ResolvedType = list, Type = list.GetType() });
```

And particularly `Type = list.GetType()`.  Seems to make perfect sense from a user's point of view.

Reply:
However, it should not matter, since the execution would ignore the `Type` property as long as `ResolvedType` was set.  So the fix here is a valid fix.

Reply:
Understood, finally...